server/store.py

```diff
@@ -121,42 +121,33 @@ def load_index(self) -> bool:
     async def initialize_or_update(self, source: Source):
         """Initializes or updates a vector store from given source"""
 
-        # If index does not already exist, create it.
-        if not self.ready():
+        # If an index does not yet exist, create it.
+        create_index = not self.ready()
```

Comment:
I realized most of the two paths are actually identical and we don't need really need multiple calls to a `generate_index()` method. All we really need is a boolean for a relevant status update and a final index creation op, and the rest utilizes the same code path.